test/e2e/install_test.go

```diff
@@ -123,9 +128,14 @@ var _ = Describe("Operator Install", func() {
 
 					bd := rukpakv1alpha1.BundleDeployment{}
 					g.Expect(c.Get(ctx, types.NamespacedName{Name: operatorName}, &bd)).To(Succeed())
-					g.Expect(bd.Status.Conditions).To(HaveLen(2))
-					g.Expect(bd.Status.Conditions[0].Reason).To(Equal("UnpackSuccessful"))
-					g.Expect(bd.Status.Conditions[1].Reason).To(Equal("InstallationSucceeded"))
+					for _, condition := range bd.Status.Conditions {
+						switch conditionType := condition.Type; conditionType {
+						case "HasValidBundle":
+							g.Expect(condition.Reason).To(Equal("UnpackSuccessful"))
+						case "Installed":
+							g.Expect(condition.Reason).To(Equal("InstallationSucceeded"))
+						}
+					}
```

Comment:
Sorry to be a bother, but if there are no conditions set, this will still pass, and I'm not sure that's what we want either?

It seems like we may want to do this:
```go
hasValidBundle := meta.FindStatusCondition(bd.Status.Conditions, rukpakv1alpha1.TypeHasValidBundle)
g.Expect(hasValidBundle).ToNot(BeNil())
g.Expect(hasValidBundle.Reason).To(Equal(rukpakv1alpha1.ReasonUnpackSuccessful))
```

And then a similar thing for the `Installed` condition.